### DISCUSSION TITLE
debug console messages are not printing in console

### DISCUSSION THREAD

**Discussion by Shashi0838** [eclipse-theia/theia#13400] [Q&A]
Hi Team,

I'm trying to send message to debug console using debugConsoleSession.appendLine and debugConsoleSession.logOutput methods,
both are not displaying in the console.

debugConsoleSession.items has the content but its not visible. 

![image](https://github.com/eclipse-theia/theia/assets/9738642/b170117a-87b5-40c3-8d29-cbffb9ff9c8e)

any idea?

Thanks in advance.

**Comment by msujew** [comment]
@Shashi0838 Can you provide a reproducible example? The API generally seems to work, since I'm seeing debug output when trying to debug JavaScript code.

**Comment by Shashi0838** [comment]
I dont know the reproducible example. I'm using vscode cpp plugin as i'm using cpp program.
and Is there any example how to inject the debugconsolesession with debugger to get output ?

Its working for me with Theia 0.8.0, now i'm trying to update to latest platform facing issue with messages

**Reply by msujew** [reply]
> and Is there any example how to inject the debugconsolesession with debugger to get output ?

No, example as far as I know, you will need to take a look at [`@theia/debug`](https://github.com/eclipse-theia/theia/tree/master/packages/debug).

> Its working for me with Theia 0.8.0, now i'm trying to update to latest platform facing issue with messages

Version 0.8? That's been released 5 years ago.

> I dont know the reproducible example. I'm using vscode cpp plugin as i'm using cpp program.

You mean the [cpptools extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools)? We don't offer support for the extension, since you're not allowed to install it into Theia due to Microsoft's proprietary license terms. See [here](https://marketplace.visualstudio.com/items/ms-vscode.cpptools/license).

**Reply by Shashi0838** [reply]
I have missed to assign consolesessionmanger to debugconsolesession. and i'm not using cpptools extesion, i'm using cpp plugin which is in the example project.
